sdk/eventhub/Azure.Messaging.EventHubs/stress/src/Configurations/TestConfiguration.cs

```diff
@@ -45,4 +45,46 @@ internal class TestConfiguration
     /// </summary>
     ///
     public int DurationInHours = 150;
+
+    /// <summary>
+    ///   Gets the partition count of the Event Hub being used for this test run.
+    /// </summary>
+    ///
+    /// <returns>The number of partitions in the Event Hub associated with this test.</returns>
+    ///
+    public async Task<int> GetEventHubPartitionCountAsync()
+    {
+        int partitionCount;
+
+        if (string.IsNullOrEmpty(EventHubsConnectionString) || string.IsNullOrEmpty(EventHub))
+        {
+            return 0;
+        }
+
+        await using (var producerClient = new EventHubProducerClient(EventHubsConnectionString, EventHub))
+        {
+            partitionCount = (await producerClient.GetEventHubPropertiesAsync().ConfigureAwait(false)).PartitionIds.Length;
+        }
+
+        return partitionCount;
+    }
+
+    /// <summary>
+    ///   Gets the list of partitions from the Event Hub being used for this test run.
+    /// </summary>
+    ///
+    /// <returns>An array holding the partition IDs associated with the Event Hub used for this test.>/returns>
+    ///
+    public async Task<string[]> GetEventHubPartitionKeysAsync()
```

Comment:
```suggestion
    public async Task<string[]> GetEventHubPartitionsAsync()
```

We probably want to be cautious here; there's a big difference between a "partition key" and a "partition id" conceptually.